### PR TITLE
add paging to pp dashboard

### PR DESCRIPTION
**Pull request by rgforsyth**
## What does this pull request do?

Adds paging to the pp dashboard.

## What is the intent behind these changes?

Part of a mitigation against dashboard timeouts.

### CODE REVIEW

File: server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts [hunk 1]
```diff
@@ -45,32 +45,41 @@ export default class ProbationPractitionerReferralsController {
...
 
   private async showDashboard(
+    req: Request,
     res: Response,
     getSentReferralsFilterParams: GetSentReferralsFilterParams,
     dashboardType: PPDashboardType
   ) {
-    const cases = await this.interventionsService.getSentReferralsForUserToken(
+    const pageNumber = req.query.page
+    const paginationQuery = {
```
[the remark this answers is not in the document]
Comment (sldblog) [reply server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts:70]: could the page size come from an environment variable config, please?

File: server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts [hunk 2]
```diff
@@ -45,32 +45,41 @@ export default class ProbationPractitionerReferralsController {
...
       res.locals.user.token.accessToken,
-      getSentReferralsFilterParams
+      getSentReferralsFilterParams,
+      paginationQuery
     )
 
-    const dedupedInterventionIds = Array.from(new Set(cases.map(referral => referral.referral.interventionId)))
+    const dedupedInterventionIds = Array.from(new Set(cases.content.map(referral => referral.referral.interventionId)))
     const interventions = await Promise.all(
       dedupedInterventionIds.map(id => this.interventionsService.getIntervention(res.locals.user.token.accessToken, id))
```
Comment (sldblog) [inline server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts:84]: _not related to this PR_
I wonder if it would make sense to pull everything at once instead of making N number of authorised calls

File: server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.test.ts [hunk 3]
```diff
@@ -132,9 +134,10 @@ describe('GET /probation-practitioner/dashboard', () => {
           },
         }),
       ]
+      const page = pageFactory.pageContent(referrals).build() as Page<SentReferral>
 
       interventionsService.getIntervention.mockResolvedValue(intervention)
-      interventionsService.getSentReferralsForUserToken.mockResolvedValue(referrals)
+      interventionsService.getSentReferralsForUserTokenPaged.mockResolvedValue(page)
```
Comment (sldblog) [inline server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.test.ts:140]: Is it worth testing that the pagination parameter (`page`) is correct passed through, with edge cases like missing = default value, string = what happens, etc?

File: server/config.ts [hunk 4]
```diff
@@ -113,6 +113,15 @@ export default {
         deadline: 22000,
       },
       agent: new AgentConfig(),
+      apiClientId: get('API_CLIENT_ID', 'interventions', requiredInProduction),
```
[the remark this answers is not in the document]
Comment (rgforsyth) [reply server/config.ts]: Good spot. No it shouldn't

File: server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts [hunk 5]
```diff
@@ -45,32 +45,46 @@ export default class ProbationPractitionerReferralsController {
...
     res: Response,
     getSentReferralsFilterParams: GetSentReferralsFilterParams,
-    dashboardType: PPDashboardType
+    dashboardType: PPDashboardType,
+    pageSize: string
   ) {
-    const cases = await this.interventionsService.getSentReferralsForUserToken(
+    const pageNumber = req.query.page
+    const paginationQuery = {
+      page: pageNumber ? Number(pageNumber) : undefined,
```
Comment (tomsmyers) [inline server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts:76]: the logic here is "if page is undefined, keep it as undefined. otherwise convert to Number, ignoring resulting NaNs". is this correct?

**Review by tomsmyers — APPROVED** [review]
nice work
